test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -1000,4 +1001,32 @@ class LazyListTest {
       assertEquals(s"mkString 3 $i", goal(3,i), precyc(3,i).mkString)
     }
   }
+  @Test
+  def tapEach: Unit = {
+    // LazyLists over Iterators
+    {
+      val lb = ListBuffer[Int]()
+      val ll = LazyList.from(Iterator(1, 2, 3, 4, 5)).tapEach(lb += _)
+      assertEquals(ListBuffer[Int](), lb)
+      assertEquals(Vector(1, 2), ll.take(2).to(Vector))
+      assertEquals(ListBuffer[Int](1, 2), lb)
+      assertEquals(4, ll(3))
+      assertEquals(ListBuffer[Int](1, 2, 3, 4), lb)
+      assertEquals(Vector(1,2,3,4,5), ll.to(Vector))
+      assertEquals(ListBuffer[Int](1, 2, 3, 4, 5), lb)
+    }
+
+    // LazyLists over Vectors
+    {
+      val lb = ListBuffer[Int]()
+      val ll = LazyList.from(Vector(1, 2, 3, 4, 5)).tapEach(lb += _)
+      assertEquals(ListBuffer[Int](), lb)
+      assertEquals(Vector(1, 2), ll.take(2).to(Vector))
+      assertEquals(ListBuffer(1, 2), lb)
+      assertEquals(4, ll(3))
+      assertEquals(ListBuffer(1, 2, 3, 4), lb) // this should change to Listbuffer(1,2,4) if LazyList becomes independently lazy
```

Comment:
you can make the `LazyList` independently lazy by doing `LazyList.tabulate(5)(_ + 1)`